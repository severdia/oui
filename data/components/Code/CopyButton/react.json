{"description":"","displayName":"CopyButton","methods":[{"name":"supportsCopyAPI","docblock":null,"modifiers":[],"params":[],"returns":null}],"props":{"code":{"type":{"name":"string"},"required":true,"description":"The code that will be copied"},"testSection":{"type":{"name":"string"},"required":false,"description":"Hook for automated JavaScript tests"}},"private":true}