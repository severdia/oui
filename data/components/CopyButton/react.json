{"description":"","methods":[],"props":{"testSection":{"type":{"name":"string"},"required":false,"description":"Hook for automated JavaScript tests"},"text":{"type":{"name":"string"},"required":true,"description":"The text or code that will be copied"}},"private":true}