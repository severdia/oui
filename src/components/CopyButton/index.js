import React from 'react';
import PropTypes from 'prop-types';
import { CopyToClipboard } from 'react-copy-to-clipboard';

import Button from '../Button';
import Icon from 'react-oui-icons';

/**
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 * @private
 */

const CopyButton = ({
  style,
  testSection,
  textToCopy,
}) => {

  return (
    <CopyToClipboard
      text={ textToCopy }>
      <Button
        style={ style === 'none' ? null : style }
        ariaLabel="Copy code snippet"
        testSection={ testSection ? `${testSection}-copy-button` : null }>
        <Icon name='clipboard' />
      </Button>
    </CopyToClipboard>
  );
};

CopyButton.propTypes = {
  /** Style option for the button */
  style: PropTypes.oneOf([
    'highlight',
    'danger',
    'danger-outline',
    'outline',
    'outline-reverse',
    'plain',
    'toggle',
    'underline',
    'unstyled',
    'none',
  ]),
  /** Hook for automated JavaScript tests */
  testSection: PropTypes.string,
  /** The text or code that will be copied */
  textToCopy: PropTypes.string.isRequired,
};

CopyButton.defaultProps = {
  style: 'plain',
};

export default CopyButton;
