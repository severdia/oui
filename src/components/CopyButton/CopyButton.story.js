import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs, select } from '@storybook/addon-knobs';

import CopyButton from '../CopyButton';

const code = `const CopyButton = ({ testSection, onClick, text }) => {
  return (
    <CopyToClipboard
      text={ textToCopy }
      onCopy={ onClick }>
      <Button
        style="plain"
        ariaLabel="Copy code snippet"
        testSection={ testSection }>
        <Icon name='clipboard' />
      </Button>
    </CopyToClipboard>
  );
}`;

const stories = storiesOf('CopyButton', module);
stories
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div id="root-preview">
      {story()}
    </div>
  ));

stories
  .add('Default', (() => {
    return (
      <CopyButton textToCopy={ code } />
    );
  }))
  .add('With style specified', (() => {
    return (
      <CopyButton
        textToCopy={ code }
        style={ select('style', {
          'none': 'none (use default button style)',
          'highlight': 'highlight',
          'danger': 'danger',
          'danger-outline': 'danger-outline',
          'outline': 'outline',
          'outline-reverse': 'outline-reverse',
          'plain': 'plain',
          'toggle': 'toggle',
          'underline': 'underline',
          'unstyled': 'unstyled',
        }, 'none') }
      />
    );
  }));
