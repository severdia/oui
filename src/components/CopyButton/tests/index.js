import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CopyButton from '../index';

describe('components/CopyButton', () => {
  it('should render children the icon button', () => {
    const component = shallow(
      <CopyButton textToCopy={ 'let bingo = "BINGO"' } testSection="fake-test-section"/>
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });

});

