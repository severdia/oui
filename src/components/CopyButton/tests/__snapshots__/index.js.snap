// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CopyButton should render children the icon button 1`] = `
<CopyToClipboard
  text="let bingo = \\"BINGO\\""
>
  <Button
    ariaLabel="Copy code snippet"
    isDisabled={false}
    isLink={false}
    isLoading={false}
    isSubmit={false}
    loadingText=""
    onBlur={[Function]}
    onClick={[Function]}
    onMouseDown={[Function]}
    style="plain"
    testSection="fake-test-section-copy-button"
    width="default"
  >
    <Icon
      name="clipboard"
    />
  </Button>
</CopyToClipboard>
`;
